src/azul/indexer/document.py

```diff
@@ -423,13 +423,8 @@ def translate_fields(cls,
                 # FIXME: Assert that a non-list field_type implies a non-list
                 #        doc (only possible for contributions).
                 #        https://github.com/DataBiosphere/azul/issues/2689
-                # FIXME: Samples are an exception since they are polymorphic.
-                #        Unused fields are left as None, even if field_types
-                #        would normally dictate otherwise.
-                #        https://github.com/DataBiosphere/azul/issues/2071
                 assert (isinstance(doc, list)
-                        or (doc is not None if forward else doc != NullableString.null_string)
-                        or path[:2] == ('contents', 'samples'))
+                        or (doc is not None if forward else doc != NullableString.null_string))
```

Comment:
Hopefully you should be able to change this to just `assert isinstance(doc, list)`. If not, there might be bigger problems, either with your PR, or mine which introduced this change.

Reply:
Also, for the record, it looks like there was a typo with the original assert. It should have read
```
 assert (isinstance(doc, list)
                        or (doc is not None if forward else doc != NullableString.null_string)
                        AND path[:2] == ('contents', 'samples'))
```

Reply:
Ah yes. The 'or' confused me into thinking the other part of the assert was related to other FIXME nearby.
Indeed the assert now passes with `assert isinstance(doc, list)`.